src/State.hs

```diff
@@ -1286,6 +1286,31 @@ deleteMessage new = do
   cId <- use csCurrentChannelId
   when (postChannelId new == cId) updateViewed
 
+maybePostUsername :: ChatState -> Post -> T.Text
+maybePostUsername st p = case do { uId <- postUserId p; name <- usernameForUserId uId st; return name } of
+    Nothing -> T.empty
+    Just x -> x
```

Comment:
Some style notes:
* The binding of (and then `return` of) `name` is redundant.
* This whole thing can be shortened to:
```
fromMaybe T.empty $ do
  uId <- postUserId p
  usernameForUserId uId st
```